internal/cli/base.go

```diff
@@ -325,6 +325,22 @@ func (c *baseCommand) Init(opts ...Option) error {
 	// config parsing, as precedence order means we take the most specific value
 	// which is the -project flag
 	if c.flagProject != "" {
+		// Warn if the project from config and the project from flags conflict
+		if c.refProject != nil && c.flagProject != c.refProject.Project {
+			c.ui.Output(warnProjectFlagMismatch, c.refProject.Project, c.flagProject, terminal.WithWarningStyle())
+
+			// The config we parsed is for some other project, so we don't want to use it
+			c.cfg = nil
+			c.refProject = nil
+			c.refApps = nil
```

Comment:
It feels a little bit messy to undo the refs that were set in the above config-parsing block - if someday we set more refs up there, we'll have to remember to roll them back here.

Moving this flag mismatch check into the config parsing block feels messy too though.

Reply:
Feels like you could move this into the `if !baseCfg.NoConfig` block because thats the only place we populate refProject. From there, it feels cleaner to avoid ever setting these extra fields. That way the place where its set is totally just one place.

Reply:
Hmm, i'm not sure if I entirely follow - we'll still need to set `refProject` outside of `baseCfg.NoConfig`, because there may be commands that need a project ref but don't want config parsed. I took another stab at it though, moving more logic up into the `baseCfg.NoConfig` block to avoid setting the refs in the first place if the config and flag defined project names collide.

Still feels a bit messy, because now we're checking `c.flagProject` twice - but it's not too bad. Let me know if you see a better pathway.